editors/code/src/config.ts

```diff
@@ -46,7 +46,11 @@ export class Config {
 
     public userConfigChanged() {
         const config = vscode.workspace.getConfiguration('rust-analyzer');
+        
+        Server.highlighter.removeHighlights();
```

Comment:
:+1:

Reply:
There are (or were) a couple of bugs where the highlighting doesn't get refreshed after changing the settings, and it's mildly annoying, so I was happy to see that.

Reply:
Ah yeah, I noticed that to.
I think removing everything is a bit drastic, there are a few parts we can cache and reuse. 
Not to mention it might be unnecessary to constantly remove it on every config update.

I didn't want to change *TOO* much, but I would have loved adding the event object to the callback and detect which portions of config changes. If that's alright with the owners, I'd add that to.

Reply:
I imagine config updates are pretty rare, and while it look a little janky, it's hardly the largest issue of RA.

Reply:
Fair enough but I had other plans here which relies on the config.